pallet/account-migration/src/lib.rs

```diff
@@ -93,15 +93,23 @@ pub mod pallet {
 		+ darwinia_staking::Config
 	{
 		/// Override the [`frame_system::Config::RuntimeEvent`].
-		type RuntimeEvent: From<Event> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 	}
 
 	#[allow(missing_docs)]
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
-	pub enum Event {
+	pub enum Event<T: Config> {
 		/// An account has been migrated.
-		Migrated { from: AccountId32, to: AccountId20 },
+		Migrated {
+			from: AccountId32,
+			to: AccountId20,
+			account_ring: Box<AccountData<Balance>>,
+			vested_ring: Option<Vec<VestingInfo<Balance, BlockNumber>>>,
+			deposited_ring: Option<Vec<Deposit>>,
+			account_kton: Option<AssetAccount>,
+			staking_ledger: Box<Option<LedgerForEvent>>,
```

Comment:
Why not use `Ledger` from the staking pallet directly, then we can avoid the unsafe part.

Reply:
Ledger requires `T: Config`. We can't pass the outside runtime type to it.